plugins/osdn/osdn.go

```diff
@@ -71,32 +74,27 @@ func (oi *OsdnRegistryInterface) CreateSubnet(nodeName string, sub *osdnapi.Subn
 	return err
 }
 
-func (oi *OsdnRegistryInterface) WatchSubnets(receiver chan *osdnapi.SubnetEvent, stop chan bool) error {
-	subnetEventQueue := oscache.NewEventQueue(cache.MetaNamespaceKeyFunc)
-	listWatch := &cache.ListWatch{
-		ListFunc: func() (runtime.Object, error) {
-			return oi.oClient.HostSubnets().List()
-		},
-		WatchFunc: func(resourceVersion string) (watch.Interface, error) {
-			return oi.oClient.HostSubnets().Watch(resourceVersion)
-		},
-	}
-	cache.NewReflector(listWatch, &api.HostSubnet{}, subnetEventQueue, 4*time.Minute).Run()
+func (oi *OsdnRegistryInterface) WatchSubnets(receiver chan<- *osdnapi.SubnetEvent, ready chan<- bool, start <-chan string, stop <-chan bool) error {
+	subnetEventQueue, reflector := oi.runEventQueue(&api.HostSubnet{}, nil)
+	sendWatchReadiness(reflector, ready)
+	startVersion := getStartVersion(start, "HostSubnet")
 
+	checkCondition := true
 	for {
 		eventType, obj, err := subnetEventQueue.Pop()
 		if err != nil {
 			return err
 		}
+		hs := obj.(*api.HostSubnet)
+		if checkCondition && skipObject(hs.ObjectMeta.ResourceVersion, startVersion) {
+			continue
+		}
+		checkCondition = false
```

Comment:
Another repeated pattern (everything from "subnetEventQueue, reflector := ..." to "checkCondition = false".